### PR TITLE
Dogfood ComInterfaceGenerator in crossgen2

### PR DESCRIPTION
**Pull request by jtschuster**
Replace the manual ComWrappers implementation in crossgen2 with the ComInterfaceGenerator implementation. 

ComInterfaceGenerator does not support chars yet, so they are replaced with ushort for the interface parameters, and manually cast to chars after the COM call.

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by agocke — APPROVED** [review]
Can't tell you how glad I am to see my code being deleted

**Comment by jkoritzinsky** [comment]
Marking this No Merge for now until we update it based on the fact that the APIs are now in the product and not in Ancillary.Interop.

**Comment by mangod9** [comment]
Is this blocked on another PR which adds new APIs so this change could use them?

**Comment by AaronRobinsonMSFT** [comment]
> Is this blocked on another PR which adds new APIs so this change could use them?

Yep. As soon as we update the repo to the shipping preview 4? (or is it 5?) SDK we can merge this.

**Comment by jtschuster** [comment]
Looks like we're on preview 4 now, @jkoritzinsky are we good to merge?

File: src/coreclr/tools/aot/ILCompiler.Diagnostics/ILCompiler.Diagnostics.csproj [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\ILCompiler.TypeSystem\ILCompiler.TypeSystem.csproj" />
+    <ProjectReference Include="..\..\..\..\libraries\System.Runtime.InteropServices\tests\Ancillary.Interop\Ancillary.Interop.csproj" />
```
Comment (jkoritzinsky) [inline src/coreclr/tools/aot/ILCompiler.Diagnostics/ILCompiler.Diagnostics.csproj:24]: We should be able to remove these two ProjectReferences now that we're on the p4 SDK

**Review by jkoritzinsky — APPROVED** [review]
One change, then LGTM!

**Pull request merged**